### PR TITLE
Bringing tutorials up to date - round two

### PR DESCRIPTION
**Pull request by johnwhitington**
This PR deals with the "Language Features" section.

When looking at the diff, it seems as if there are several wholesale deletions and additions. Most are in fact simply movements of content between files.

### CODE REVIEW

File: site/learn/tutorials/functional_programming.md [hunk 1]
```diff
@@ -47,10 +47,7 @@ function as one of its arguments.
 So far so simple. If you're familiar with C/C++ then this looks like
 passing a function pointer around. Java has some sort of abomination
 called an anonymous class which is like a dumbed-down, slow and
```
Comment (avsm) [inline site/learn/tutorials/functional_programming.md]: I feel that this entire sentence about Java and the language used here is unnecessary and could be removed along with the Perl removal.

File: site/learn/tutorials/null_pointers_asserts_and_warnings.md [hunk 2]
```diff
@@ -38,18 +38,15 @@ special SQL `NULL` value. SQL ignores this automatically when you ask it
...
 
 Programming languages also support nulls, although they may be easier to
-use in some than in others. In Perl, any scalar (ie. number or string)
-can be `undef` (Perl's way of saying null). This is a cause of many
-warnings which are often ignored by inexperienced programmers, even
-though they probably represent serious errors. In Java, any reference to
-an object can also be null, so it might make sense in Java to store the
+use in some than in others. In Java, any reference to
+an object can be null, so it might make sense in Java to store the
 age as an `Integer` and allow references to the age to be `null`. In C
```
Comment (avsm) [inline site/learn/tutorials/null_pointers_asserts_and_warnings.md]: null is both verbatim and not in the same sentence -- perhaps pick one style for it.

**Review by avsm — APPROVED** [review]
Just a couple of minor suggestions but otherwise lgtm!

**Comment by johnwhitington** [comment]
Both fixed. Thanks.

**Comment by avsm** [comment]
Thanks!

**Pull request merged**